app/routes.py

```diff
@@ -1 +1,275 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, abort, make_response, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import os
+import requests
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"details":"Invalid data"}, 400))
+
+    model = cls.query.get(model_id)
+    
+    if not model:
+        abort(make_response({"details": f"{cls.__name__} {model_id} is not found"}, 404))
+    
+    return model
+
+def validate_request(cls, request):
+    request_body = request.get_json()
+
+    if not request_body.get("title"):
+        abort(make_response({"details":"Invalid data"}, 400))
+    
+    elif not request_body.get("description") and cls == Task:
```

Comment:
We could write the conditions for this logic in one line like this: 
```python
if not (request_body.get('title') or request_body.get('description')):
```

Reply:
We could also generalize your `validate_request` function to check for any request body like this:
```python
def validate_request_body(request_body, keys):
    for key in keys:
        if not request_body.get(key): 
            abort(make_response({
                'Invalid Data': f'missing key: {key}'
            }, 400))

    return True
```
We can pass in the `request_body` and a list of strings that are `keys` and then check to see if those keys are present.